### PR TITLE
Python fixes 2

### PR DESCRIPTION
**Pull request by EWouters**
Second pass of Python fixes with the [Trunk](https://trunk.io/) linters.

Not sure if the exception handling in aac3d501a655489b92cea96d7a19b10767cf8d26 will lead to the same behavior.

There are some warnings left with respect to exception handling, but I will leave them for now.

### CODE REVIEW

File: backend/chainlit/callbacks.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 @trace
 def password_auth_callback(
-    func: Callable[[str, str], Awaitable[Optional[User]]]
+    func: Callable[[str, str], Awaitable[Optional[User]]],
```
Comment (dokterbob) [inline backend/chainlit/callbacks.py:19]: Why the `,` here?

File: backend/chainlit/data/sql_alchemy.py [hunk 2]
```diff
@@ -7,7 +7,6 @@
 
 import aiofiles
 import aiohttp
-
```
Comment (dokterbob) [inline backend/chainlit/data/sql_alchemy.py:10]: Why remove this whitespace? I feel it's often indicative to separate:
1. Python stdlib imports
2. Third-party deps
3. Internal deps

File: backend/chainlit/discord/__init__.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
-try:
-    import discord
-except ModuleNotFoundError:
+import importlib.util
+
+if importlib.util.find_spec("discord") is None:
     raise ValueError(
         "The discord package is required to integrate Chainlit with a Slack app. Run `pip install discord --upgrade`"
```
Comment (dokterbob) [inline backend/chainlit/discord/__init__.py:5]: Unrelated, but did someone forget to change Slack to Discord here? @willydouhard

File: backend/chainlit/step.py [hunk 4]
```diff
@@ -140,7 +140,7 @@ def sync_wrapper(*args, **kwargs):
                     try:
                         if result and not step.output:
                             step.output = result
-                    except:
+                    except Exception as e:
                         step.is_error = True
                         step.output = str(e)
```
Comment (dokterbob) [inline backend/chainlit/step.py:145]: Wow, was this a broken exception handler here?

File: backend/tests/data/conftest.py [hunk 5]
```diff
@@ -1,7 +1,6 @@
-import pytest
-
 from unittest.mock import AsyncMock
 
+import pytest
```
Comment (dokterbob) [inline backend/tests/data/conftest.py:3]: Why was this changed?

File: backend/tests/data/test_sql_alchemy.py [hunk 6]
```diff
@@ -1,17 +1,13 @@
...
-
-from chainlit.data.base import BaseDataLayer, BaseStorageClient
+from chainlit import User
+from chainlit.data.base import BaseStorageClient
 from chainlit.data.sql_alchemy import SQLAlchemyDataLayer
 from chainlit.element import Text
-from chainlit import User
-from chainlit.user import PersistedUser
+from sqlalchemy import text
+from sqlalchemy.ext.asyncio import create_async_engine
```
Comment (dokterbob) [inline backend/tests/data/test_sql_alchemy.py:10]: See earlier comments on preferred import order. If we're not following that, I'd like to hear your suggestion.
It's a good time to start getting consistent on this (more important than 'my own way'). :)

**Review by dokterbob — COMMENTED** [review]
There was a lot of changes to import order.

Without those, apart from my few comments/questions, I think I'm happy to merge.

With those, I'd like to first agree on how we sanitise imports from now on, making whatever we stick to a policy so I won't have to deal with reviewing import statement reorders again. ;)

My personal preference is:
1. Stdlib imports (as modules, so `import datatime` over `from datetime import datetime`.
2. 3rd party deps
3. Internal deps.

Beyond that, I guess alphabetical sorting is nice. But none of that is essential. Consistency is. :)

File: backend/chainlit/callbacks.py (hunk 1 above)
Comment (EWouters) [reply backend/chainlit/callbacks.py:19]: That is the result of running `ruff format`.

File: backend/chainlit/step.py (hunk 4 above)
Comment (EWouters) [reply backend/chainlit/step.py:145]: Indeed, this was introduced in [4e0baab](https://github.com/Chainlit/chainlit/commit/4e0baabe3fd48abdd2f9998395b02c40e1f3d9e4#diff-4677e8359efad4a24ef6e70a44fd94c6549e46d017c3a0a9692bfd7d890bb75eR115).

File: backend/chainlit/data/sql_alchemy.py (hunk 2 above)
Comment (EWouters) [reply backend/chainlit/data/sql_alchemy.py:10]: That is the result of running `ruff format`.

File: backend/tests/data/test_sql_alchemy.py (hunk 6 above)
Comment (EWouters) [reply backend/tests/data/test_sql_alchemy.py:10]: That is the result of running the Trunk formatter, which uses ruff, isort and black to organize the imports. I'm trying to move everything to be consistent. It is not my own way.

File: backend/tests/data/conftest.py (hunk 5 above)
Comment (EWouters) [reply backend/tests/data/conftest.py:3]: That is the result of running the Trunk formatter.

**Comment by EWouters** [comment]
Thanks for your reply. For all the whitespace and import changes I used formatters to make the changes. None of these are my own preference. Moving to consistency is the goal.

You can pick which formatter you like, for instance:
- `ruff check --select I --fix`
- `isort .`
- `black .`

Trunk uses a combination of these. With the current changes all files pass the trunk linters. If you rather choose one of the above, I can change the imports to the order of that formatter.

File: backend/chainlit/discord/__init__.py (hunk 3 above)
Comment (EWouters) [reply backend/chainlit/discord/__init__.py]: Yes, unrelated, but fixed in c9aaa4b9d247a4435056495ae1d86ea6f565f180.

**Comment by dokterbob** [comment]
Thanks for the feedback! 

I think what I missed, specifically, is having Python stdlib imports first, then other imports and finally internal/project imports.

AFAIK `ruff check --select I --fix` does this (but please check, it might be that we need to update settings in `pyproject.toml`.

Looking forward to fixups in that as well as resolution of the merge conflict in the SQLAlchemy test.

Once again, for the future; (several) smaller PR's (e.g. only doing one of these things) are easier to review and thus faster to merge (and less merge conflicts, as well). :)

**Comment by dokterbob** [comment]
@EWouters Didn't want to pass you by but also wanted to 'clean house' in terms of PR's, so I rebased your PR minus formatting (prefer to first have clear standards, enforced on a CI-level before heading that way).
